docs/en/docs/ecosystem/udf/java-user-defined-function.md

```diff
@@ -185,6 +187,131 @@ CREATE AGGREGATE FUNCTION simple_sum(INT) RETURNS INT PROPERTIES (
     "type"="JAVA_UDF"
 );
 ```
+
+```JAVA
+package org.apache.doris.udf.demo;
+
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.logging.Logger;
+
+/*UDAF for calculating the median*/
+public class MedianUDAF {
+    Logger log = Logger.getLogger("MedianUDAF");
+
+    // State storage
+    public static class State {
+        // Precision of the result
+        int scale = 0;
+        // Whether this is the first time to execute add() for the data under a certain aggregation condition of a certain tablet
+        boolean isFirst = true;
+        //Data storage
+        public StringBuilder stringBuilder;
+    }
+
+    //State initialization
+    public State create() {
+        State state = new State();
+        //Pre-initialize based on the amount of data to be aggregated for each aggregation condition under each tablet, for improved performance
+        state.stringBuilder = new StringBuilder(1000);
+        return state;
+    }
+
+
+    // Handle the data for each unit under each aggregation condition for each tablet
+    public void add(State state, Double val, int scale) {
+        try {
+            if (val != null && state.isFirst) {
+                state.stringBuilder.append(scale).append(",").append(val).append(",");
+                state.isFirst = false;
```

Comment:
the scale is seems always 0?

Reply:
No, isFirst controls adding the scale parameter to the stringBuilder when add is executed for the first time, and I have verified this through practice.